ouroboros-network/src/Ouroboros/Network/PeerSelection/LedgerPeers.hs

```diff
@@ -268,97 +267,100 @@ ledgerPeersThread :: forall m peerAddr resolver exception.
                      , Exception exception
                      , Ord peerAddr
                      )
-                  => StdGen
-                  -> DNSSemaphore m
-                  -> (IP.IP -> Socket.PortNumber -> peerAddr)
-                  -> Tracer m TraceLedgerPeers
-                  -> STM m UseLedgerPeers
-                  -> LedgerPeersConsensusInterface m
-                  -> DNSActions resolver exception m
+                  => PeerActionsDNS peerAddr resolver exception m
+                  -> WithLedgerPeersArgs m
+                  -- blocking request for ledger peers
                   -> STM m (NumberOfPeers, LedgerPeersKind)
-                  -- ^ a blocking action which receives next request for more
-                  -- ledger peers
+                  -- response with ledger peers
                   -> (Maybe (Set peerAddr, DiffTime) -> STM m ())
                   -> m Void
-ledgerPeersThread inRng dnsSemaphore toPeerAddr tracer readUseLedgerAfter
-                  ledgerPeersConsensusInterface dnsActions
-                  getReq putRsp = do
-    go inRng (Time 0) Map.empty Map.empty
+ledgerPeersThread PeerActionsDNS {
+                    paToPeerAddr,
+                    paDnsActions,
+                    paDnsSemaphore }
+                  WithLedgerPeersArgs {
+                    wlpRng,
+                    wlpConsensusInterface,
+                    wlpTracer,
+                    wlpGetUseLedgerPeers }
```

Comment:
At some point, we should experiment with `DuplicateRecrodFields`; there have been some improvements there which would allow us to not use the silly prefixes.